### PR TITLE
algorithm: improve PassiveAggressiveRegressor

### PR DESCRIPTION
**Pull request by amo33**
This PR updates PassiveAggressiveRegressor.
Output using ```make test```:
```
      Start  7: RegressionTest.LinearRegression
 7/10 Test  #7: RegressionTest.LinearRegression ..............   Passed    0.01 sec
      Start  8: RegressionTest.PolynomialRegression
 8/10 Test  #8: RegressionTest.PolynomialRegression ..........   Passed    0.01 sec
 ```
 
 Test Log:
 ```
 7/10 Testing: RegressionTest.LinearRegression
7/10 Test: RegressionTest.LinearRegression
Command: "/home/flatflow/build/flatflow/scheduler/internal/algorithm/algorithm_test" "--gtest_filter=RegressionTest.LinearRegression"
Directory: /home/flatflow/build/flatflow/scheduler/internal/algorithm
"RegressionTest.LinearRegression" start time: May 16 10:17 UTC
Output:
----------------------------------------------------------
Running main() from /home/flatflow/third_party/googletest/googletest/src/gtest_main.cc
Note: Google Test filter = RegressionTest.LinearRegression
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from RegressionTest
[ RUN      ] RegressionTest.LinearRegression
[       OK ] RegressionTest.LinearRegression (6 ms)
[----------] 1 test from RegressionTest (6 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (6 ms total)
[  PASSED  ] 1 test.
<end of output>
Test time =   0.01 sec
----------------------------------------------------------
Test Passed.
"RegressionTest.LinearRegression" end time: May 16 10:17 UTC
"RegressionTest.LinearRegression" time elapsed: 00:00:00
----------------------------------------------------------

8/10 Testing: RegressionTest.PolynomialRegression
8/10 Test: RegressionTest.PolynomialRegression
Command: "/home/flatflow/build/flatflow/scheduler/internal/algorithm/algorithm_test" "--gtest_filter=RegressionTest.PolynomialRegression"
Directory: /home/flatflow/build/flatflow/scheduler/internal/algorithm
"RegressionTest.PolynomialRegression" start time: May 16 10:17 UTC
Output:
—————————————————————————————
Running main() from /home/flatflow/third_party/googletest/googletest/src/gtest_main.cc
Note: Google Test filter = RegressionTest.PolynomialRegression
[==========] Running 1 test from 1 test suite.
[—————] Global test environment set-up.
[—————] 1 test from RegressionTest
[ RUN      ] RegressionTest.PolynomialRegression
[       OK ] RegressionTest.PolynomialRegression (7 ms)
[—————] 1 test from RegressionTest (7 ms total)

[—————] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (7 ms total)
[  PASSED  ] 1 test.
<end of output>
Test time =   0.01 sec
—————————————————————————————
Test Passed.
"RegressionTest.PolynomialRegression" end time: May 16 10:17 UTC
"RegressionTest.PolynomialRegression" time elapsed: 00:00:00
—————————————————————————————
 ```

### CODE REVIEW

File: flatflow/scheduler/internal/algorithm/regression.h [hunk 1]
```diff
@@ -33,62 +33,143 @@ namespace algorithm {
...
+  }
+
+  double intercept = 0.0;
+
+ protected:
+  double epsilon_;
+  std::vector<double> coef_;
+  std::vector<double> power_;
+  static constexpr auto C_ = 1.0;
+  static constexpr auto max_iter_ = 1000;
```
Comment (9rum) [inline flatflow/scheduler/internal/algorithm/regression.h:93]: The type of ``max_iter_`` should be ``std::size_t``.
That is, ``static constexpr auto max_iter_ = static_cast<std::size_t>(1000);``.

File: flatflow/scheduler/internal/algorithm/regression.h [hunk 2]
```diff
@@ -33,62 +33,143 @@ namespace algorithm {
...
 
  protected:
   double epsilon_;
+  double intercept_;
   std::vector<double> coef_;
   std::vector<double> power_;
-  static constexpr auto kMaxIter_ = 1000;
-  static constexpr auto kC_ = 1.0;
+  static constexpr auto C_ = 1.0;
+  static constexpr auto max_iter_ = 1000;
```
Comment (9rum) [inline flatflow/scheduler/internal/algorithm/regression.h:172]: Same as line 93.

File: flatflow/scheduler/internal/algorithm/regression_test.cc [hunk 3]
```diff
@@ -20,95 +20,122 @@
...
+  static constexpr auto kLinear = static_cast<std::size_t>(1);
+  static constexpr auto kQuadratic = static_cast<std::size_t>(2);
+  static constexpr auto kEpsilon = static_cast<double>(0.0001);
+  static constexpr auto kThreshold = static_cast<double>(0.01);
+  static constexpr auto kNode = static_cast<int>(4);
+  static constexpr auto kMinRange = static_cast<int>(100);
+  static constexpr auto kMaxRange = static_cast<int>(200);
+  static constexpr auto kMinCoef = static_cast<int>(300);
+  static constexpr auto kMaxCoef = static_cast<int>(400);
+  static constexpr std::size_t kDatasetSize = 100;
```
Comment (9rum) [inline flatflow/scheduler/internal/algorithm/regression_test.cc:99]: For consistency, define ``kDatasetSize`` with ``static_cast``:
i.e., ``static constexpr auto kDatasetSize = static_cast<std::size_t>(100);``

**Review by 9rum — CHANGES_REQUESTED** [review]
@amo33 Thanks for filing this PR.
It seems you have implemented one of the most important parts pretty well.
I have noted a few minor fixes, so please let me know after change.

**Comment by amo33** [comment]
@9rum Thanks for your detailed review. 
I have updated the code to address all the changes you suggested:
Let me know if any further changes are needed.

**Review by 9rum — APPROVED** [review]
LGTM

**Pull request merged**